validation/src/main/scala/hmda/validation/rules/lar/validity/V300.scala

```diff
@@ -0,0 +1,66 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.census.CBSATractLookup
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.Result
+import hmda.validation.rules.EditCheck
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+import hmda.validation.dsl.PredicateRegEx._
+
+object V300 extends EditCheck[LoanApplicationRegister] {
+
+  val cbsaTracts = CBSATractLookup.values
+
+  val validCombination = cbsaTracts.map { cbsa =>
+    (cbsa.geoIdMsa, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }
+
+  val validMdCombination = cbsaTracts.map { cbsa =>
+    (cbsa.metDivFp, cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }
+
+  val validStateCountyCombination = cbsaTracts.map { cbsa =>
+    (cbsa.state, cbsa.county, cbsa.tractDecimal)
+  }
```

Comment:
I'd suggest experimenting with `.toSet` even so.  It'll likely lead to faster lookup times, as you don't have to search the whole list.  See the changes in `ef8167c775d6df76d3721b4d9bcf69e347e93ddd` for examples.

On that note, I notice that we have a bunch of similar lookup methods happening to the `CBSATractLookup` values, and scattered around the rules.  I wonder whether there's something common and useful to extract.  Doesn't necessarily have to be in this PR, but seems worth considering.